internal/monitors/collectd/kong/kong.go

```diff
@@ -157,12 +164,172 @@ type Config struct {
 	StatusCodesBlacklist []string `yaml:"statusCodesBlacklist"`
 }
 
+// PythonConfig returns the embedded python.Config struct from the interface
+func (c *Config) PythonConfig() *python.Config {
+	return c.pyConf
+}
+
 // Monitor is the main type that represents the monitor
 type Monitor struct {
-	collectd.MonitorCore
+	python.PyMonitor
 }
 
 // Configure configures and runs the plugin in collectd
-func (rm *Monitor) Configure(conf *Config) error {
-	return rm.SetConfigurationAndRun(conf)
+func (m *Monitor) Configure(conf *Config) error {
+	conf.pyConf = &python.Config{
+		MonitorConfig: conf.MonitorConfig,
+		Host:          conf.Host,
+		Port:          conf.Port,
+		ModuleName:    "kong_plugin",
+		ModulePaths:   []string{filepath.Join(os.Getenv(constants.BundleDirEnvVar), "plugins", "collectd", "kong")},
+		TypesDBPaths:  []string{filepath.Join(os.Getenv(constants.BundleDirEnvVar), "plugins", "collectd", "types.db")},
+		PluginConfig: map[string]interface{}{
+			"URL": conf.URL,
+		},
+	}
+
+	if conf.Verbose != nil {
+		conf.pyConf.PluginConfig["Verbose"] = *conf.Verbose
+	}
+	if conf.Name != "" {
+		conf.pyConf.PluginConfig["Name"] = conf.Name
+	}
+	if conf.AuthHeader != nil {
+		conf.pyConf.PluginConfig["AuthHeader"] = []string{conf.AuthHeader.HeaderName, conf.AuthHeader.Value}
+	}
+	if conf.VerifyCerts != nil {
+		conf.pyConf.PluginConfig["VerifyCerts"] = *conf.VerifyCerts
+	}
+	if conf.CABundle != "" {
+		conf.pyConf.PluginConfig["CABundle"] = conf.CABundle
+	}
+	if conf.ClientCert != "" {
+		conf.pyConf.PluginConfig["ClientCert"] = conf.ClientCert
+	}
+	if conf.ClientCertKey != "" {
+		conf.pyConf.PluginConfig["ClientCertKey"] = conf.ClientCertKey
+	}
+	if conf.IntervalSeconds > 0 {
+		conf.pyConf.PluginConfig["Interval"] = conf.IntervalSeconds
+	}
+	if len(conf.Metrics) > 0 {
+		values := make([][]interface{}, 0, len(conf.Metrics))
+		for _, m := range conf.Metrics {
+			values = append(values, []interface{}{m.MetricName, m.ReportBool})
+		}
+		conf.pyConf.PluginConfig["Metric"] = map[string]interface{}{
+			"#flatten": true,
+			"values":   values,
+		}
+	}
+	if conf.ReportAPIIDs != nil {
+		conf.pyConf.PluginConfig["ReportAPIIDs"] = *conf.ReportAPIIDs
+	}
+	if conf.ReportAPINames != nil {
+		conf.pyConf.PluginConfig["ReportAPINames"] = *conf.ReportAPINames
+	}
+	if conf.ReportServiceIDs != nil {
+		conf.pyConf.PluginConfig["ReportServiceIDs"] = *conf.ReportServiceIDs
+	}
+	if conf.ReportServiceNames != nil {
+		conf.pyConf.PluginConfig["ReportServiceNames"] = *conf.ReportServiceNames
+	}
+	if conf.ReportRouteIDs != nil {
+		conf.pyConf.PluginConfig["ReportRouteIDs"] = *conf.ReportRouteIDs
+	}
+	if conf.ReportHTTPMethods != nil {
+		conf.pyConf.PluginConfig["ReportHTTPMethods"] = *conf.ReportHTTPMethods
+	}
+	if conf.ReportStatusCodeGroups != nil {
+		conf.pyConf.PluginConfig["ReportStatusCodeGroups"] = *conf.ReportStatusCodeGroups
+	}
+	if conf.ReportStatusCodes != nil {
```

Comment:
Since `nil` isn't a valid value for collectd plugin config values, why don't we just use `nil` to mean not present and then you can just put all of these pointers into the `pluginConfig` literal map above and make this a lot cleaner.

If the python runner doesn't already, it will need to filter out nil values.  This should be safe to do since you can't put `nil`/`null` into a python plugin config anyway according to https://collectd.org/documentation/manpages/collectd-python.5.shtml#config.